src/app/models/Task.ts

```diff
@@ -0,0 +1,6 @@
+export interface Task {
+    name: string;
+    id?: string;
```

Comment:
why do you have optional parameters? your model should be well defined and immutable (readonly)